### ISSUE TITLE
[iOS] New errors after updating today

### ISSUE THREAD

**Issue by slothstronaut** [j3k0/cordova-plugin-purchase#966]
# !!! PLEASE USE IN SUBJECT [IOS] OR [ANDROID] AS PREFIX FOR YOUR ISSUE !!!

### system info

Please report your OS, OS version, Compiling System and Version, used Plugin Version and any helpful information that can help us to resolve the problem.

In example:
```
OSX Catalina
Cordova 9.0
Device: Any iOS Device
Plugin Version: 10.0.0
```

### Expected behavior

Build without errors

### Observed behavior

Several new build errors after updating:

Use of undeclared identifier 'SKProductDiscountTypeIntroductory'
Use of undeclared identifier 'SKProductDiscountTypeSubscription'

cordova-plugin-purchase/InAppPurchase.m:394:41: Unknown receiver 'SKPaymentDiscount'; did you mean 'SKProductDiscount'?

InAppPurchase.m:153:18: Use of undeclared identifier 'SKProductDiscountTypeIntroductory'

No visible @interface for 'SKProductDiscount' declares the selector 'initWithIdentifier:keyIdentifier:nonce:signature:timestamp:'

https://www.dropbox.com/s/vmy6v8t44mngv98/Screenshot%202019-12-09%2007.16.56.png?dl=0

### Steps to reproduce

Update plugin to 10.0 and try to build

**Comment by slothstronaut** [comment]
This was fixed by reverting back to 9.0.1, i.e.:

```
cordova plugin remove cordova-plugin-purchase
cordova plugin add cordova-plugin-purchase@9.0.1 --variable BILLING_KEY="XXX"
```

After reverting, build works fine again.

**Comment by j3k0** [comment]
Two reasons I see:

- Are you using an old version of XCode?
- Is your app compiled against an old version of iOS SDK?

**Comment by oduverne** [comment]
HI,
I have the same issue with online Adobe Phonegap Build:
/project/My Cake Calculator/Plugins/cordova-plugin-purchase/InAppPurchase.m:153:18: error: use of undeclared identifier 'SKProductDiscountTypeIntroductory'
            case SKProductDiscountTypeIntroductory: return @"Introductory";
                 ^
/project/My Cake Calculator/Plugins/cordova-plugin-purchase/InAppPurchase.m:154:18: error: use of undeclared identifier 'SKProductDiscountTypeSubscription'
            case SKProductDiscountTypeSubscription: return @"Subscription";
                 ^
/project/My Cake Calculator/Plugins/cordova-plugin-purchase/InAppPurchase.m:394:21: error: property 'paymentDiscount' not found on object of type 'SKMutablePayment *'
            payment.paymentDiscount = [[SKPaymentDiscount alloc]
                    ^
/project/My Cake Calculator/Plugins/cordova-plugin-purchase/InAppPurchase.m:394:41: error: unknown receiver 'SKPaymentDiscount'; did you mean 'SKProductDiscount'?
            payment.paymentDiscount = [[SKPaymentDiscount alloc]

No error on Android

**Comment by slothstronaut** [comment]
> Two reasons I see:
> 
> * Are you using an old version of XCode?
> * Is your app compiled against an old version of iOS SDK?

Yes, running XCode 10.1 and iOS 12.1, not the latest version but this is the only plugin that has issues, and the version I'm using is only a year old. I'm concerned if I upgrade XCode & iOS 12.1 only for this plugin it may break others. Its already very difficult to get all plugins to agree. Pretty big risk and not easy to revert.

**Comment by j3k0** [comment]
Indeed, latest version of the plugin requires iOS 12.2 SDK (that became
available with XCode 10.2).

The new Discount Offers API for iOS has been required by some users (so you can
offer discounts to your subscribers). But unfortunately this requires iOS 12.2
SDK support (the plugin still work on older devices, it does runtime checks).

I'll mention that in the release notes.

**Comment by rafaellop** [comment]
@j3k0 I also have some issues with the 10.0.1 on iOS and I'm using the newest xcode and latest SDKs. I'm unable to debug the issue because it happens in the production version only. Purchasing works in the sandboxied version on a test device, but after the app was released I noticed no purchases are possible because of my flow which worked perfectly with the previous version of the plugin.

I use the `store.ready()` function to set own variable which helps my app to know the init was successful and store is available. It seems that it is never set to true when in the production and the result is no purchase is possible. My app checks the variable and finds the store is not initialized and displays error. 

Reverting back to 9.x.
